### DISCUSSION TITLE
Welcome to OpenVPN-over-Proxy Discussions!

### DISCUSSION THREAD

**Discussion by Null3rror** [Null3rror/OpenVPN-over-Proxy#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by caothemanh** [comment]
Install error on ubuntu
Welcome to ARMBIAN 5.88 user-built Ubuntu 18.04.5 LTS 5.1.0-aml-s905
System load:   0.07 0.07 0.07   Up time:       3:03 hours
Memory usage:  10 % of 1806MB   IP:            192.168.1.8
CPU temp:      47°C
Usage of /:    10% of 29G

[ 0 security updates available, 35 updates total: apt upgrade ]
Last check: 2021-02-14 01:27

[ General system configuration (beta): armbian-config ]

Last login: Sun Feb 14 03:27:11 2021 from 171.253.132.185

manh@aml:~$ sudo wget https://raw.githubusercontent.com/Null3rror/OpenVPN-over-Proxy/master/openvpn-over-proxy-install.sh -O openvpn-over-proxy-install.sh && sudo bash openvpn- over- proxy -install.sh
[sudo] password for manh:
--2021-02-14 04:29:43--  https://raw.githubusercontent.com/Null3rror/OpenVPN-over-Proxy/master/openvpn-over-proxy-install.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.110.133, 185.199.108.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 29369 (29K) [text/plain]
Saving to: 'openvpn-over-proxy-install.sh'

openvpn-over-proxy-i 100%[=====================>]  28.68K  --.-KB/s    in 0.007s

2021-02-14 04:29:45 (4.00 MB/s) - 'openvpn-over-proxy-install.sh' saved [29369/29369]

bash: openvpn-: No such file or directory
manh@aml:~$

**Reply by Null3rror** [reply]
Hi there, thanks for trying out my code :)
Seems like you are entering the wrong command, there shouldn't be spaces between  
``` sudo bash openvpn- over- proxy -install.sh ```
if you remove the spaces, it should be like this: 
``` sudo bash openvpn-over-proxy-install.sh ```

Hope it helps.

**Reply by caothemanh** [reply]
Thanks for your feedback, it worked. excuse me for asking a problem. Where I stay they configured the file as attachment and ran fine but I don't know how they did it.[Openvpn](https://drive.google.com/file/d/1HIhNVerpOzi1ZZuIlPRoH03oaXmuwrWA/view?usp=drivesdk)

**Reply by caothemanh** [reply]
if I leave # in the configuration it doesn't work, it doesn't seem to be compatible.

**Reply by Null3rror** [reply]
Glad it helped.

> if I leave # in the configuration it doesn't work, it doesn't seem to be compatible.

I can't understand, where do you leave the #?

**Reply by caothemanh** [reply]
client
dev tun
proto tcp
http-proxy 171.227.10.201 3128
remote 171.227.10.201 1194
resolv-retry infinite
http-proxy-option VERSION 1.1
#http-proxy-option CUSTOM-HEADER ""
#http-proxy-option CUSTOM-HEADER "GET http://mobion.vn HTTP/1.1"
http-proxy-option CUSTOM-HEADER Host mobion.vn
http-proxy-option CUSTOM-HEADER X-Forward-Host mobion.vn
http-proxy-option CUSTOM-HEADER X-Forwarded-For mobion.vnvn
http-proxy-option CUSTOM-HEADER Referrer mobion.vn
nobind
persist-key
persist-tun
remote-cert-tls server
auth SHA512
cipher AES-256-CBC
ignore-unknown-option block-outside-dns
block-outside-dns
verb 3
<ca>

**Reply by caothemanh** [reply]
if I omit the # sign in the above client configuration file, it can't work. where I live, They do so in the client configuration file to make it work.

**Reply by caothemanh** [reply]
I don't understand how they installed openvpn server with http proxy, when configuring client configuration with option. http-proxy-option CUSTOM-HEADER "" http-proxy-option CUSTOM-HEADER "GET http://mobion.vn HTTP / 1.1"

**Reply by Null3rror** [reply]
I think it's best if you ask this in OpenVPN forum and please don't show your IP address since you're bound to get attacks.

**Reply by caothemanh** [reply]
thanks i will change ip

**Comment by ahmedmar3y** [comment]
error connect openvpn gui windows  recv_line: TCP port read failed on recv(): Unknown error (errno=10054)
 pleas help

**Reply by Null3rror** [reply]
Hi Ahmed,
Let's debug it on discord, add me so we can talk:
RunTime_Terror#3482

**Comment by breeze90** [comment]
hey guys! does it still work? Fro example in China in 2023?
